crates/validation/src/materialization.rs

```diff
@@ -327,7 +327,22 @@ async fn walk_materialization(
     {
         shard_template.id.clone()
     } else {
-        assemble::shard_id_prefix(pub_id, materialization, labels::TASK_TYPE_MATERIALIZATION)
+        assemble::shard_id_prefix(
+            match endpoint {
```

Comment:
optional nit: feel free to disagree, but I found this inlined match expression to be a little hard on the eyes, and would I think prefer a variable.